src-built-in/components/UIAPI/hooks/linker.ts

```diff
@@ -1,24 +1,92 @@
 import { useDispatch, useSelector } from 'react-redux';
 import { useEffect } from "react";
-
-import { RootState, Linker } from '../types';
-import * as actions from '../actions/linkerActions';
+import { RootState, LinkerState, actions, Channel, Channels } from '../types';
 
 // Encapsulate the linker initialization and Redux dispatch functions inside the hook
 export const useLinker = () => {
+
     const dispatch = useDispatch();
-    const linker: Linker = useSelector((state: RootState) => state.linker);
+    const state: LinkerState = useSelector((state: RootState) => state.linker);
 
     useEffect(() => {
-		dispatch(actions.init());
+
+        FSBL.Clients.Logger.system.log("Linker component is mounted. Initializing.");
+        finsembleWindow.addEventListener("blurred", hideWindow);
+        finsembleWindow.addEventListener("shown", showWindow);
+        setInitialChannels();
+        initAccessibleLinkerMode();
+        FSBL.Clients.RouterClient.addResponder("Finsemble.LinkerWindow.SetActiveChannels", onActiveChannelsChanged);
+
         return () => {
-            dispatch(actions.cleanUp());
+            FSBL.Clients.Logger.system.log("Linker component is unmounted. Cleaning up the event listeners.");
+            finsembleWindow.removeEventListener("blurred", hideWindow);
+            finsembleWindow.removeEventListener("shown", showWindow);
+            FSBL.Clients.RouterClient.removeEventListener("Finsemble.LinkerWindow.SetActiveChannels", onActiveChannelsChanged);
+        }
+    }, []);
+
+    useEffect(() => {
+        FSBL.Clients.WindowClient.fitToDOM();
+    }, [state])
+
+    const hideWindow = () => {
+        finsembleWindow.hide();
+    };
+    const showWindow = () => {
+        finsembleWindow.focus();
+    };
+
+    const setInitialChannels = () => {
```

Comment:
Would prefer these to be defined before they're called. Just a code organization thing. I may move them if you aren't working today.